src/Authenticator/JwtAuthenticator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Authenticator;
+
+use Authentication\Identifier\IdentifierCollection;
+use Authentication\Result;
+use Cake\ORM\Entity;
+use Firebase\JWT\JWT;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use RuntimeException;
+use stdClass;
+
+class JwtAuthenticator extends TokenAuthenticator {
+
+    /**
+     * @inheritDoc
+     */
+    protected $_defaultConfig = [
+        'header' => 'Authorization',
+        'queryParam' => 'token',
+        'tokenPrefix' => 'bearer',
+        'allowedAlgs' => ['HS256'],
+        'entityClass' => Entity::class,
+        'returnPayload' => true,
+        'key' => null,
+        'salt' => null
```

Comment:
Having a `salt` key is unnecessary. Just having `key` is enough and it user does not provide any value for it then use value of `Security::salt()` as fallback if `Cake\Utility\Security` is available else throw an exception. The fallback check and throwing exception can be done in `_decodeToken()` itself.